mindmeld/components/dialogue.py

```diff
@@ -1011,6 +1027,38 @@ async def call_async(self, request, responder):
         """
         self(request, responder)
 
+    def invoke(self, request, responder):
+        """
+        Invoke slot-filling as a direct call without requiring a decorator.
+        """
+        self._invoked = True
+        kwargs = {'domain': request.domain, 'intent': request.intent}
+
+        # Set dialogue rule to auto_fill and
+        # store current rule for resetting once invoke is complete
+
+        name = self._entrance_handler.__name__
+        _called_from = sys._getframe().f_back.f_code.co_name
+
+        responder.frame["_called_from"] = _called_from
+        responder.frame[
+            "previous_rule"
+        ] = self._app.app_manager.dialogue_manager.handler_map[_called_from]
+
+        try:
+            self._app.app_manager.dialogue_manager.add_dialogue_rule(name, self.__call__, **kwargs)
+        except AssertionError:
+            self._app.app_manager.dialogue_manager.handler_map[name] = self.__call__
```

Comment:
What does the exception case do?

Reply:
It bypasses the exception in adding new dialogue rules and updates the rule regardless.

Reply:
Is it possible to never collide with an existing handler? What if the name was unique?

Reply:
This is for the case, when they restart the same flow. If the user calls say 'transfer' twice, this flow needs to be called again, and possibly the sub-flow again. This causes that exception.

Reply:
We should add documentation to this exception

Reply:
In the case where the user says "transfer" twice, do we need to explicitly remap the function? I think it is still the same function right?

Reply:
I tried out a case where the same secondary handler could be used by two different AutoFill functions. This causes the mapping to shift, if I'm not wrong. So just ensuring that it is mapped for every invoke.